force-app/main/default/lwc/rd2Service/model.js

```diff
@@ -52,15 +79,71 @@ const DEFAULT_INITIAL_STATE = {
     isChangeLogEnabled: null,
     periodToYearlyFrequencyMap: null,
     closedStatusValues: [],
-    defaultInstallmentPeriod: null // new!
+    defaultInstallmentPeriod: null, // new!
+
+    //Permissions
+    InstallmentPeriodPermissions: {},
+    InstallmentFrequencyPermissions: {}
+};
+
+const isRecurringTypeChanged = (state) => {
+    return state.recurringType !== initialViewState.recurringType;
+};
+
+const isNewRecord = (state) => {
+    return !state.recordId;
+};
+
+const getAnnualValue = (state) => {
+    const { donationValue, recurringFrequency, recurringPeriod, periodToYearlyFrequencyMap } = state;
+    const yearlyFrequency = periodToYearlyFrequencyMap[recurringPeriod];
+
+    return donationValue * (yearlyFrequency / recurringFrequency);
+};
+
+const getFixedValue = (state) => {
+    const { donationValue, paidAmount, paidInstallments, plannedInstallments } = state;
+    const remainingInstallments = plannedInstallments - paidInstallments;
+
+    return paidAmount + (remainingInstallments * donationValue);
+};
+
+const getDonationValue = (state) => {
+    if (state.recurringType === RECURRING_TYPE_OPEN) {
+        return getAnnualValue(state);
+    } else if (state.recurringType === RECURRING_TYPE_FIXED) {
+        return getFixedValue(state);
+    }
 }
 
+const getChangeType = (state) => {
```

Comment:
I still need to test this (in main to compare, and in this PR), but wouldn't this override a manually entered value for Change Type? We don't want to update Change Type if it was set by the User.

Reply:
It would override Change Type but only as a result of editing a field that influences change type (Frequency/Period/Amount/Number of Installments).